src/soter/openssl/soter_asym_cipher.c

```diff
@@ -109,11 +110,8 @@ soter_status_t soter_asym_cipher_cleanup(soter_asym_cipher_t* asym_cipher)
 
 	if (asym_cipher->pkey_ctx)
 	{
-		EVP_PKEY* pkey = EVP_PKEY_CTX_get0_pkey(asym_cipher->pkey_ctx);
-		EVP_PKEY_CTX_free(asym_cipher->pkey_ctx);
-		if(pkey){
-		    EVP_PKEY_free(pkey);
-		}
+        EVP_PKEY_CTX_free(asym_cipher->pkey_ctx);
+        asym_cipher->pkey_ctx = NULL;
```

Comment:
- indentation
- can you confirm that `EVP_PKEY_CTX_free` also frees "contained" `pkey`?

Reply:
https://github.com/openssl/openssl/blob/master/crypto/evp/pmeth_lib.c#L339
and also checked with valgrind (he warned that there was double free when I leave `EVP_PKEY_free(pkey)`)